src/components/app/WindowTitle.js

```diff
@@ -70,16 +70,25 @@ class WindowTitleImpl extends PureComponent<Props> {
           const { meta } = profile;
           let title = '';
           if (formattedMetaInfoString) {
-            title += formattedMetaInfoString + SEPARATOR;
+            title += formattedMetaInfoString;
           }
-          title += _formatDateTime(
-            meta.startTime + (meta.profilingStartTime || 0)
-          );
+
+          // Print the startTime only if it's provided.
+          if (meta.startTime > 0) {
+            title +=
+              SEPARATOR +
+              _formatDateTime(meta.startTime + (meta.profilingStartTime || 0));
+          }
+
           if (dataSource === 'public') {
             title += ` (${dataSource})`;
           }
 
-          title += SEPARATOR + PRODUCT;
+          if (title !== '') {
+            // Add the separator only if we added some information before.
+            title += SEPARATOR;
+          }
```

Comment:
Instead of this, you could build up a titleFragments array and then do an array join at the end, but this is fine too.

Reply:
Ah yeah, I guess I chose to do it this way because it was the current implementation. Your suggestion is cleaner, but when I was thinking about changing the code, I realized that `if (dataSource === 'public') {` branch doesn't really fit that implementation fully, because it doesn't join on the left side with the`SEPARATOR`. See ``title += ` (${dataSource})`;`` and the space in front.

So doing `titleFragments.join(SEPARATOR)` would work for the others except this. We can make it work by appending the last string, like ``titleFragments[titleFragments.length - 1] += ` (${dataSource})`;`` but that also makes it a bit dirty like the current implementation.

I gues we can discuss about changing that ` (public)` to something like ` – Public –`. But not so sure about it.

Reply:
Yeah I think that's why I did it this way at the time :-)